### ISSUE TITLE
Compiling Error: error: reference to ‘wait’ is ambiguous

### ISSUE THREAD

**Issue by Andres6936** [alandefreitas/matplotplusplus#16]
```shellsession
andres6936@Tierra ~/CLionProjects/matplotplusplus/cmake-build-debug $ uname -a
Linux Tierra 4.15.0-46-generic #49~16.04.1-Ubuntu SMP Tue Feb 12 17:45:24 UTC 2019 x86_64 GNU/Linux
```

```shellsession
andres6936@Tierra ~/CLionProjects/matplotplusplus/cmake-build-debug $ cmake --version
cmake version 3.16.0
```

```shellsession
andres6936@Tierra ~/CLionProjects/matplotplusplus/cmake-build-debug $ gcc --version
gcc (Ubuntu 9.3.0-10ubuntu2~16.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
```

**_The following is CMake terminal output._**

<details>
<summary>Execution of CMake Command ..</summary>

```shellsession
andres6936@Tierra ~/CLionProjects/matplotplusplus/cmake-build-debug $ cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMAKE_MODULE_PATH=/home/andres6936/CLionProjects/matplotplusplus/cmake
-- CPM: adding package nodesoup@0.1 (4bcd8cd2d6c963ccf053f8c19dbb63c065467527)
-- CPM: adding package cimg@0.221 (v.221)
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so (found version "80") 
JPEG Found
-- Found TIFF: /usr/lib/x86_64-linux-gnu/libtiff.so (found version "4.0.6") 
TIFF Found
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.8") 
ZLIB Found
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.2.54") 
PNG Found
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Could NOT find BLAS (missing: BLAS_LIBRARIES) 
-- LAPACK requires BLAS
-- A library with LAPACK API not found. Please specify library location.
-- Could NOT find BLAS (missing: BLAS_LIBRARIES) 
-- Could NOT find FFTW (missing: FFTW_INCLUDE_DIRS) 
-- Checking for module 'OpenEXR'
--   No package 'OpenEXR' found
-- Checking for module 'Magick++'
--   No package 'Magick++' found
-- Found X11: /usr/include   
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Looking for C++ include filesystem
-- Looking for C++ include filesystem - found
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED
-- Performing Test CXX_FILESYSTEM_NO_LINK_NEEDED - Success
-- Looking for __fbufsize
-- Looking for __fbufsize - found
-- Looking for min
-- Looking for min - not found
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so   
ERROR: EXTERNAL LIBRARY 'GLAD' NOT FOUND
-- Found GLFW3: /usr/local/lib/libglfw3.a  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/andres6936/CLionProjects/matplotplusplus/cmake-build-debug
```
</details>

```shellsession
andres6936@Tierra ~/CLionProjects/matplotplusplus/cmake-build-debug $ make -j2
```

```cxx
Scanning dependencies of target generate_examples
Scanning dependencies of target nodesoup
[  1%] Linking CXX static library libnodesoup.a
[  1%] Linking CXX executable generate_examples
[  4%] Linking CXX static library libmatplot.a
Scanning dependencies of target example_plot_12
Scanning dependencies of target example_plot_2
[  4%] Building CXX object examples/line_plot/plot/CMakeFiles/example_plot_12.dir/plot_12.cpp.o
[  4%] Building CXX object examples/line_plot/plot/CMakeFiles/example_plot_2.dir/plot_2.cpp.o
/home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_2.cpp: In function ‘int main()’:
/home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_2.cpp:11:5: error: reference to ‘wait’ is ambiguous
   11 |     wait();
      |     ^~~~
In file included from /usr/include/stdlib.h:42,
                 from /usr/include/c++/9/cstdlib:75,
                 from /usr/include/c++/9/bits/stl_algo.h:59,
                 from /usr/include/c++/9/algorithm:62,
                 from /home/andres6936/CLionProjects/matplotplusplus/source/matplot/util/common.h:8,
                 from /home/andres6936/CLionProjects/matplotplusplus/source/matplot/matplot.h:9,
                 from /home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_2.cpp:1:
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:66:7: note: candidates are: ‘union wait’
   66 | union wait
      |       ^~~~
In file included from /home/andres6936/CLionProjects/matplotplusplus/source/matplot/matplot.h:9,
                 from /home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_2.cpp:1:
/home/andres6936/CLionProjects/matplotplusplus/source/matplot/util/common.h:125:10: note:                 ‘void matplot::wait()’
  125 |     void wait();
      |          ^~~~
/home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_12.cpp: In function ‘int main()’:
/home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_12.cpp:10:5: error: reference to ‘wait’ is ambiguous
   10 |     wait();
      |     ^~~~
In file included from /usr/include/stdlib.h:42,
                 from /usr/include/c++/9/bits/std_abs.h:38,
                 from /usr/include/c++/9/cmath:47,
                 from /home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_12.cpp:1:
/usr/include/x86_64-linux-gnu/bits/waitstatus.h:66:7: note: candidates are: ‘union wait’
   66 | union wait
      |       ^~~~
In file included from /home/andres6936/CLionProjects/matplotplusplus/source/matplot/matplot.h:9,
                 from /home/andres6936/CLionProjects/matplotplusplus/examples/line_plot/plot/plot_12.cpp:2:
/home/andres6936/CLionProjects/matplotplusplus/source/matplot/util/common.h:125:10: note:                 ‘void matplot::wait()’
  125 |     void wait();
      |          ^~~~

```

**Comment by alandefreitas** [comment]
I had no idea this `union wait` existed as a global function. Does changing it to `matplot::wait();` solves the issue?

I don't know what the best solution for that would be to avoid this conflict. I would change it to `pause()`, but it seems like there's also a global `pause` function. 😒

**Comment by bthcode** [comment]
> Does changing it to matplot::wait();

that fixes it

**Comment by alandefreitas** [comment]
That should have fixed it then: 9730e6062d3f5ca6ef90d37861deaf6f3c10aa26
